projects/rero/ng-core/src/lib/record/search/record-search.component.ts

```diff
@@ -1116,9 +1109,35 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
     if (recordsAggregation.doc_count != null) {
       aggregation.doc_count = recordsAggregation.doc_count;
     }
+    aggregation.value.buckets.map(bucket => this.processBuckets(bucket, aggregation.key));
     aggregation.loaded = true;
   }
 
+  /**
+   * Enrich the bucket with several properties: indeterminate, parent, aggregationKey.
+   *
+   * @param bucket elastic bucket to process
+   * @param aggregationKey bucket parent key
+   * @returns the indeterninate state of the bucket
+   */
+  processBuckets(bucket, aggregationKey): boolean {
+    // checkbox indeterminate state
+    bucket.indeterminate = false;
+    for (const k in bucket) {
+      if (bucket[k] && bucket[k].buckets) {
+        for (const childBucket of bucket[k].buckets) {
+          // store the parent: usefull to remove parent filters
+          childBucket.parent = bucket;
+          // store the aggregation key as we re-organize the bucket structure
+          bucket.aggregationKey = aggregationKey;
+          bucket.indeterminate ||= this._recordSearchService.hasFilter(k, childBucket.key);
+          bucket.indeterminate = this.processBuckets(childBucket, k) || bucket.indeterminate;
```

Comment:
you should revert condition to fasten processing.
If `bucket.indeterminate` is already set to `True` by previous line, then no need to process the childBucket. Isn't it ?